File: pkg/model/api_key.go
package model

import "time"

// APIKeyPrefix 是 Key 的固定前缀。带前缀是为了让它在日志、粘贴板、代码库里
// 一眼可辨——扫描工具也能靠它命中误提交。
const APIKeyPrefix = "ebx_"

// TenantAPIKey 是租户的对外取件凭据，一个租户至多一把。
type TenantAPIKey struct {
	TenantID string
	// TokenHash 是 SHA-256 摘要，鉴权时按它查。
	TokenHash string
	// TokenEnc 是明文的 AES-GCM 密文，只用于页面回显（见 000012 迁移的说明）。
	TokenEnc string
	// Scopes 使用空格分隔；旧 Key 的数据库默认值只有 mail:read。
	Scopes    string
	CreatedAt time.Time
	UpdatedAt time.Time
}

// APIKeyView 是回显给页面的形态。明文只在这里出现，不进任何日志。
type APIKeyView struct {
	Token     string    `json:"token"`
	Scopes    []string  `json:"scopes"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

type APIKeyIdentity struct {
	TenantID string
	Scopes   map[string]bool
}

const (
	APIKeyScopeMailRead    = "mail:read"
	APIKeyScopeLedgerRead  = "ledger:read"
	APIKeyScopeLedgerWrite = "ledger:write"
)

File: web/src/api/apiKey.ts
import client from "@/lib/client";
import type { ApiResponse } from "@/lib/client";

// 与后端 model.APIKeyView 手工同步。明文只在这一个接口上出现。
export interface ApiKey {
  token: string;
  scopes?: string[];
  created_at: string;
  updated_at: string;
}

export const apiKeyApi = {
  // 还没生成时后端回 data:null，页面据此显示「生成」而不是「重置」。
  get: async (tenantID: string) =>
    (await client.get<ApiResponse<ApiKey | null>>(`/api/v1/tenants/${tenantID}/api-key`)).data,
  reset: async (tenantID: string) =>
    (await client.post<ApiResponse<ApiKey>>(`/api/v1/tenants/${tenantID}/api-key/reset`)).data,
};
